src/main/java/com/sqltest/demo/service/ExcelService.java

```diff
@@ -0,0 +1,119 @@
+package com.sqltest.demo.service;
+
+import com.sqltest.demo.model.SalesOrder;
+import com.sqltest.demo.repository.SalesOrderRepo;
+import org.apache.poi.ss.usermodel.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+@Service
+public class ExcelService {
+    @Autowired
+    private SalesOrderRepo salesOrderRepository;
+
+    private static final String PLACEHOLDER = "N/A";
+
+    public void readExcelAndSaveToDatabase() throws IOException {
+        ClassPathResource resource = new ClassPathResource("excel/data.xlsx");
```

Comment:
Define a specific path for the excel in application property file and fetch in class.
eg:
fms-so-dump: /fmsdir/SODUMP/

@Value("${fms-so-dump}")
	private String soDir;

Reply:
Since the Excel is in the resources folder, I have used  ResourceLoader to load it from the classpath.

Updated code:
private static final String FILE_NAME = "data.xlsx";

 Resource resource = resourceLoader.getResource("classpath:excel/" + FILE_NAME);